### DISCUSSION TITLE
Can anyone help with inconsistent behaviour on Mac backend?

### DISCUSSION THREAD

**Discussion by arblackman** [libusb/hidapi#346] [Q&A]
I'm currently trying to write a small app to switch my Logitech MX Keys Keyboard and MX Master Mouse simultaneously between connected machines.

A *LOT* of people are keen to resolve this - see these articles:
- https://github.com/Logitech/logi_craft_sdk/issues/28
- https://support.logi.com/hc/en-us/community/posts/360050755634

With the right support - this is actually a super easy task - and I've solved it for Windows, but not Mac :(

On my Windows machine I've written a small app (using this library) that listens to the USB device, and when it detects the keyboard changing, it sends a HID write message to also change the mouse - the basics are here: 

Sadly, I tried to do the same with the Mac version of the hidapi library, but there is a strange behaviour - wherein the message I'd expect to receive when switching the keyboard does not arrive, and only when I switch back to the Mac I then receive 2 messages - one being the "switch away", (which I am looking for) and one being the "switch back" - but since they only arrive together on switching back, my goal is impossible.

Can anyone suggest a way to capture this special keypress (Logitech MX Keys Keyboard).

Finally, an additional wrinkle - I cannot sudo, or use Admin.

**Comment by Youw** [comment]
> I've written a small app

Could you share a code sample with comments where it doesn't work on macOS?
